python/sglang/srt/model_executor/model_runner.py

```diff
@@ -163,7 +163,15 @@ def __init__(
         # Init OpenMP threads binding
         omp_cpuids = os.environ.get("SGLANG_CPU_OMP_THREADS_BIND", "all")
         if omp_cpuids == "all":
-            self.local_omp_cpuid = "all"
+            cpu_ids_by_node = get_cpu_ids_by_node()
+            n_numa_node = len(cpu_ids_by_node)
+
+            assert self.tp_size <= n_numa_node, (
+                f"SGLANG_CPU_OMP_THREADS_BIND is not set, in this case, "
+                f"tp_size {self.tp_size} should be smaller than number of numa node on the machine {n_numa_node}. "
+                f"If you need tp_size to be larger than number of numa node, please set the CPU cores for each tp rank via SGLANG_CPU_OMP_THREADS_BIND explicitly."
+            )
```

Comment:
in case self.tp_size != n_numa_node, give a warning say that: "detected system has N numa node available, tensor parallel set M, so use only M numa nodes."

Reply:
warning is added in https://github.com/mingfeima/sglang/pull/77/commits/402c91ac2809d30dd93eed6399ff6de2535213bf